streaming/base/storage/upload.py

```diff
@@ -155,7 +161,7 @@ def __init__(self,
             self.local = out[0]
             self.remote = out[1]
 
-        if os.path.exists(self.local) and len(os.listdir(self.local)) != 0:
+        if not exist_ok and os.path.exists(self.local) and len(os.listdir(self.local)) != 0:
```

Comment:
Are we emptying the `out` directory if `exist_ok` is `True` ? or what is behavior when `exist_ok` is `True` ? There would be some stale shard files if we don't empty it before writing right ?

Reply:
yes, this Is the most concerning part. 
with "exist_ok=True", it allows merge_index utility to create UploadClouder on an existing folder without an error is thrown. But it also introduces the risk where the user accidentally set exist_ok=True, so the stale shards are not removed before writing new shard files to it.

Reply:
Can we add a warning message for this to let users about the behavior ?

Reply:
Added.